### PR TITLE
[FIX] stock: fix multi-company rental schedule

### PR DESCRIPTION
**Pull request by ltinel**
The rental schedule (Gantt view) shows the display name of the rented products,
which is read in sudo mode (`models.py` > `_read_group_format_result`). This
display name includes the total product quantity, which considers all companies
in the DB when computed in sudo mode (and without specifying a warehouse or
location). However, we should only consider the selected companies.

This change adds a domain which limits the companies that are considered when
computing product quantities.

Enterprise PR: https://github.com/odoo/enterprise/pull/61871

opw-3877326

### CODE REVIEW

File: addons/stock/models/product.py [hunk 1]
```diff
@@ -120,7 +120,7 @@ def _compute_valid_ean(self):
     @api.depends('stock_move_ids.product_qty', 'stock_move_ids.state', 'stock_move_ids.quantity')
     @api.depends_context(
         'lot_id', 'owner_id', 'package_id', 'from_date', 'to_date',
-        'location', 'warehouse',
+        'location', 'warehouse', 'company'
```
Comment (Sehnde) [inline addons/stock/models/product.py:123]: why did you change the `depends_context` ? You are not calling `companies.ids` from the context but from the env, so it works by itself. Imo, there's no need for that.
```suggestion
        'location', 'warehouse'
```

**Review by Sehnde — CHANGES_REQUESTED** [review]
Could you also add a test, please?

File: addons/stock/models/product.py (hunk 1 above)
Comment (ltinel) [reply addons/stock/models/product.py]: Actually, there's a special treatment for the `company` key (see [1]). The cache will use the value from the `env` instead of the `context`.

However, after writing the test, I noticed that this doesn't work if multiple companies are selected (since the cache only uses the first one). As a result I used the `allowed_company_ids` key instead (which considers all selected companies).

[1] https://github.com/odoo/odoo/blob/823f30f4fd01667e37fc90fbbd795c3e5e857103/odoo/api.py#L819-L820

**Comment by ltinel** [comment]
> Could you also add a test, please?

Done :)

**Review by Sehnde — APPROVED** [review]
